pkg/adapter/containers_remote.go

```diff
@@ -276,19 +291,120 @@ func (r *LocalRuntime) CreateContainer(ctx context.Context, c *cliconfig.CreateV
 // Run creates a container overvarlink and then starts it
 func (r *LocalRuntime) Run(ctx context.Context, c *cliconfig.RunValues, exitCode int) (int, error) {
 	// TODO the exit codes for run need to be figured out for remote connections
-	if !c.Bool("detach") {
-		return 0, errors.New("the remote client only supports detached containers")
-	}
 	results := shared.NewIntermediateLayer(&c.PodmanCommand)
 	cid, err := iopodman.CreateContainer().Call(r.Conn, results.MakeVarlink())
 	if err != nil {
 		return 0, err
 	}
-	fmt.Println(cid)
 	_, err = iopodman.StartContainer().Call(r.Conn, cid)
+	if c.Bool("detach") {
+		fmt.Println(cid)
+		return 0, err
+	}
+	if err := r.attach(ctx, os.Stdin, os.Stdout, cid); err != nil {
```

Comment:
I don't know if this works - for local, we specifically order the start and attach calls so start comes after attach, to ensure we don't lose output

Reply:
it works, but im contemplating if i **can** do it the same for a remote console

Reply:
I don't really know how it works right now... Might be racy - container starts slow enough that attach has time to complete

Reply:
ok i have this working now